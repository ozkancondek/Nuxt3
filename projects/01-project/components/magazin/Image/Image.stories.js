import ImageComponent from './Image.vue'
import { STORYBOOK_UI_COMPONENTS_PREFIX } from '~/components/storybook.constants'

export default {
  title: `${STORYBOOK_UI_COMPONENTS_PREFIX}/Image`,
  component: ImageComponent,
  argTypes: {},
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const Template = (args, { argTypes }) => ({
  components: { ImageComponent },
  props: Object.keys(argTypes),
  template: '<ImageComponent v-bind="$props" v-on="$props"></ImageComponent>',
})

export const ImageExample = Template.bind({})
ImageExample.args = {
  src: '/magazinapi/dam/Freizeit/draussen-abkuehlen/mann-in-haengematte-garten.jpg',
  alt: 'placeholder',
  aspectRatio: 1.78,
  originalAspectRatio: 1,
  ratioSquare: false,
  fpX: 0.5,
  fpY: 0.5,
  fpZoom: 1,
  square: false,
}

export const ImageSquareExample = Template.bind({})
ImageSquareExample.args = {
  ...ImageExample.args,
  square: false,
}

export const ImageOriginalExample = Template.bind({})
ImageOriginalExample.args = {
  src: '/magazinapi/dam/Terrasse/winterharte-kuebelpflanzen/Sitzplatz-Gartenbank-Beet-Kuebelpflanze-Giesskanne.jpg',
  alt: 'Ein Sitzplatz mit einer Gartenbank vor einem Beet mit Kübelpflanze und Gießkanne',
  caption: '',
  captionAlignRight: false,
  aspectRatio: 1.78,
  originalAspectRatio: 1.5,
  fpX: 0.45,
  fpY: 0.55,
  fpZoom: 1,
  articleImage: true,
}
