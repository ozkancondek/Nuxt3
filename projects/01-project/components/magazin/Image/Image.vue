<template>
  <picture class="tw-flex" @click="$emit('click')">
    <source
      v-for="(source, index) in sources"
      :key="compKey + index"
      :media="source.media"
      :srcset="source.srcset"
    />
    <img
      :src="srcOriginal"
      :alt="alt"
      class="tw-w-full tw-object-cover"
      @load="onLoad"
    />
  </picture>
</template>

<script lang="ts">
import { useRuntimeConfig } from "#app";

import { PropType } from "vue";

type Source = { media: string; srcset: string };

export default defineComponent({
  name: "ImageComponent",
  props: {
    src: {
      type: String,
      default: "",
    },
    alt: {
      type: String,
      default: "",
    },
    aspectRatio: {
      type: Number,
      default: 1,
    },
    originalAspectRatio: {
      type: Number,
      default: 1,
    },
    fpX: {
      type: Number,
      default: 0.5,
    },
    fpY: {
      type: Number,
      default: 0.5,
    },
    fpZoom: {
      type: Number,
      default: 1,
    },
    widths: {
      type: Array as PropType<number[]>,
      required: false,
      default: () => [330, 690, 910, 1120],
    },
    resizeOnLoad: {
      type: Boolean,
      default: false,
    },
  },
  emits: ["click"],
  setup(props) {
    const config = useRuntimeConfig();
    const debug = config.imgixDebug === true;
    const compKey =
      "pic-source-" + (Math.random() + 1).toString(36).substring(7) + "-";
    const sources = computed<Source[]>(() => {
      const { src, fpX, fpY, fpZoom, aspectRatio } = props;
      const options = { src, fpX, fpY, fpZoom, aspectRatio, debug };

      return [];
    });
    const srcOriginal = computed<string>(() => {
      const { src, fpX, fpY, fpZoom, aspectRatio } = props;
      const options = { src, fpX, fpY, fpZoom, aspectRatio, debug };

      return src;
    });
    const onLoad = () => {
      if (props.resizeOnLoad) {
        window.dispatchEvent(new Event("resize"));
      }
    };
    return {
      srcOriginal,
      sources,
      compKey,
      onLoad,
    };
  },
});
</script>
