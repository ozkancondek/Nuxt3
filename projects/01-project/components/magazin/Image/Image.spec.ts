import { shallowMount } from '@vue/test-utils'
//import { createTestingPinia } from '@pinia/testing'
import Image from '~/components/magazin/Image/Image.vue'

vi.mock('#imports', () => {
  return {
    useRuntimeConfig() {
      return {
      
      }
    }
  }
})
test('renders Image  correctly', () => {
  //const pinia = createTestingPinia()
  const wrapper = shallowMount(Image, {
  /*   global: {
     plugins: [pinia],
    }, */ 
    mocks: {
      $t: () => {},
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})
