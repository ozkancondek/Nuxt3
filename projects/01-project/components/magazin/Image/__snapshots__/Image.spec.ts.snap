// Vitest Snapshot v1

exports[`renders Image  correctly 1`] = `
<picture
  class="tw-flex"
>
  
  
  <img
    alt=""
    class="tw-w-full tw-object-cover"
    src=""
  />
</picture>
`;
